__tests__/__packages__/cli-test-utils/src/environment/TestEnvironment.ts

```diff
@@ -102,6 +127,29 @@ export class TestEnvironment {
             const pluginDir = testEnvironment.workingDir + "/plugins";
             require("rimraf").sync(pluginDir);
         }
+
+        // Clean up resources
+        for (const file of testEnvironment.resources.localFiles) {
+            deleteLocalFile(file);
+        }
+        // Check if session exists; if not, create one
+        const session = testEnvironment.resources.session || await TestEnvironment.createSession();
+        for (const file of testEnvironment.resources.files) {
+            deleteFiles(session, file);
+        }
+        for (const job of testEnvironment.resources.jobs) {
+            deleteJob(session, job);
+        }
+        for (const jobData of testEnvironment.resources.jobData) {
+            if (jobData.jobname && jobData.jobid) {
+                deleteJobCommon(session, jobData);
+            } else {
+                Imperative.console.info('Error: Missing jobname or jobid for jobData:', jobData);
+            }
+        }
+        for (const dataset of testEnvironment.resources.datasets) {
+            deleteDataset(session, dataset);
+        }
```

Comment:
after each on of these for-loops, should we consider clearing the resource arrays? 
That way we may avoid duplicate delete operations on the same resource in case the `cleanUp` function is called more than once in a single test file 😋